### PR TITLE
feat: add EC and BLS key support to KeyCrypterAESCBC

### PR DESCRIPTION
**Pull request by HashEngineering**
* add a test that uses a EC private key for AES

## Checklist:

- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have added or updated relevant unit/integration/functional/e2e tests
- [ ] I have made corresponding changes to the documentation

**For repository code-owners and collaborators only**
- [ ] I have assigned this pull request to a milestone

### CODE REVIEW

File: core/src/main/java/org/bitcoinj/crypto/KeyCrypterAESCBC.java [hunk 1]
```diff
@@ -30,18 +33,17 @@
...
  * <p>You can use this class to:</p>
  *
- * <p>1) Using a ECDSA private key and a public key, create an AES key that can encrypt and decrypt data.
+ * <p>1) Provide an ECKey or BLSSecretKey as the AES Key.
  * </p>
  *
- * <p>2) Using the AES Key generated above, you then can encrypt and decrypt any bytes using
+ * <p>2) Using the AES Key above, you then can encrypt and decrypt any bytes using
  * the AES symmetric cipher.</p>
  */
```
Comment (HashEngineering) [inline core/src/main/java/org/bitcoinj/crypto/KeyCrypterAESCBC.java:45]: Some of the documentation here applies to derived classes.

File: core/src/main/java/org/bitcoinj/crypto/KeyCrypterAESCBC.java [hunk 2]
```diff
@@ -115,6 +117,31 @@ EncryptedData encrypt(byte[] plainBytes, byte [] iv, KeyParameter aesKey) throws
         }
     }
 
+    @Override
+    public Protos.Wallet.EncryptionType getUnderstoodEncryptionType() {
+        return Protos.Wallet.EncryptionType.ENCRYPTED_BLS_KEYEXCHANGE_AES;
+    }
```
Comment (HashEngineering) [inline core/src/main/java/org/bitcoinj/crypto/KeyCrypterAESCBC.java:123]: We don't have a correct value for this yet.  Need to fix it.

File: core/src/main/java/org/bitcoinj/crypto/KeyCrypterAESCBC.java [hunk 3]
```diff
@@ -115,6 +117,31 @@ EncryptedData encrypt(byte[] plainBytes, byte [] iv, KeyParameter aesKey) throws
...
+
+    @Override
+    public KeyParameter deriveKey(CharSequence password) throws KeyCrypterException {
+        throw new UnsupportedOperationException("use one other deriveKey methods instead");
+    }
```
Comment (HashEngineering) [inline core/src/main/java/org/bitcoinj/crypto/KeyCrypterAESCBC.java:128]: This class and derived classes do not support deriving a key from a password.  Instead this class, keys are "derived" from private keys of different formats.

So the keys are not derived, rather the byte array is wrapped inside a KeyParameter object.

Derived classes can derive a key through other means.

File: core/src/main/java/org/bitcoinj/wallet/Protos.java [hunk 4]
```diff
@@ -16488,6 +16488,10 @@ private Wallet() {
        * <code>ENCRYPTED_ECDH_KEYEXCHANGE_AES = 4;</code>
        */
       ENCRYPTED_ECDH_KEYEXCHANGE_AES(4),
+      /**
+       * <code>ENCRYPTED_AES = 5;</code>
+       */
+      ENCRYPTED_AES(5),
```
Comment (HashEngineering) [inline core/src/main/java/org/bitcoinj/wallet/Protos.java:16494]: Fortunately, with this change, we are only adding an enum value.  No changes are made to the wallet format.  

This enum is meant to specify the type of encryption used in wallet, but the values 3-5 are not used for wallet encryption at this time.

File: core/src/test/java/org/bitcoinj/crypto/KeyCrypterAESCBCTest.java [hunk 5]
```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright 2013 Jim Burton.
+ * Copyright 2014 Andreas Schildbach
```
Comment (Syn-McJ) [inline core/src/test/java/org/bitcoinj/crypto/KeyCrypterAESCBCTest.java:3]: Looks like missing Dash here?

**Review by Syn-McJ — APPROVED** [review]
Looks good

File: core/src/test/java/org/bitcoinj/crypto/KeyCrypterAESCBCTest.java (hunk 5 above)
Comment (HashEngineering) [reply core/src/test/java/org/bitcoinj/crypto/KeyCrypterAESCBCTest.java:3]: yes, you are right

**Pull request merged**